Slim/Player/Protocols/HTTP.pm

```diff
@@ -286,9 +339,59 @@ sub canDirectStream {
 	return $url;
 }
 
+# TODO: what happens if this method is overloaded in a sub-class that does not call its parents method
+sub canDirectStreamSong {
+	my ( $class, $client, $song ) = @_;
+	
+	# can't go direct if we are synced or proxy is set by user
+	my $direct = $class->canDirectStream( $client, $song->streamUrl(), $class->getFormatForURL() );
+	return 0 unless $direct;
```

Comment:
Could `$direct` hereafter ever be anything else than "truthy"? Is anything expecting this to be anything other than 0 or 1? If so, please leave a note. If not, please get rid of it and just `return 1` below instead.

Reply:
But I wanted to stop testing here if the canDirectStream tells me it cannot be direct. No need to test further and especially the tests below should not be run if the canDirectStream already said "no". I could have added a if block, but I thought it was more simple to just exit here and have less indentations